<img src = "https://raw.githubusercontent.com/bgonzalezgo/UdeCatalunya/refs/heads/main/logo_uc_grande.png" alt = "Encabezado" width = "100%">  </img>


***Especializacón en Big Data y Analítica de Datos***

*Materia: Limpieza y tranformación de datos*

**Actividad Individual – Inspección inicial de datos con Python y pandas**

***Estudiante: Brandon González Gómez***

# *Actividad a resolver*

1. Dataset oficial
Utilice el conjunto “Ventas” indicado en el listado de datasets suministrado en el siguiente
enlace
https://www.kaggle.com/datasets/alejandromontero1/ventas
Descargue el archivo CSV y colóquelo en la misma carpeta que su código, salvo que trabaje
en Google Colab (ver § 4).
2. Objetivo académico
• Cargar el CSV en un DataFrame denominado df.
• Obtener los diagnósticos estructurales y estadísticos mínimos (ver § 3).
• Practicar selección y filtrado con tres variaciones adicionales de .loc[] y .iloc[] respecto a
las que ya contiene el archivo base Archivo_Python_Ventas_Inmuebles.ipynb.
3. Requisitos de código (mínimos e ineludibles)
Paso lógico Comandos indispensables Evidencia esperada en
consola
0 – Preparación import pandas as pd Título “Preparación
completada”
1 – Carga pd.read_csv() (parámetro
extra opcional: index_col, sep
o encoding)
Mensaje con ruta y
confirmación
2 – Verificación estructural df.shape, df.columns,
df.head(), df.tail()
Salida de filas y columnas
3 – Exploración df.info(), df.describe(),
df.describe(include="object"),
df.isna().sum()
Resumen técnico y
estadístico
4 – Selección y filtrado Al menos: una selección
df['col'], otra df[[...]]; dos
ejemplos con .loc[]; dos
Impresión parcial de los
subconjuntos
ejemplos con .iloc[]; tres
variaciones adicionales
distintas.
* No se exige ningún otro componente de calidad de código (funciones, manejo de
excepciones, PEP-8, etc.).



***Se importa la liberia pandas para la manipulación y el análisis de datos***

In [1]:
import pandas as pd
print('Preparación completada')

Preparación completada


# *Verificación Estructural*

***Carga de datos***

In [ ]:
# Se descarga archivo compartido por el tutor y se carga en mi reposoritorio de GIT para poder utilizar el dataset y no tener que tenerlo descargado
# para su ejecución, si no siempre utilizar el enlace para poder usar el archivo.
url = "https://raw.githubusercontent.com/bgonzalezgo/UdeCatalunya/refs/heads/main/BD_Ventas.csv" # en la variable URL se guarda el link del dataset
df = pd.read_csv(url, index_col=False, encoding='latin2', sep=',') # Se selecciona el archivos CSV desde la URL indicando que el separador es ;

print("✅ Datos cargados exitosamente en el DataFrame 'df'.")

✅ Datos cargados exitosamente en el DataFrame 'df'.


***Muestra los primeros diez registros***

In [ ]:
df.head(10)

Year Month      Customer    Product  Units_Sold  Price_per_Unit  \
0  2019   Jan  Customer 103  Product D          93       22.426305   
1  2019   Jan   Customer 72  Product E          21       19.821771   
2  2019   Jan  Customer 467  Product C          75       87.286734   
3  2019   Jan  Customer 100  Product C          22       10.359100   
4  2019   Jan  Customer 344  Product D          30       25.172216   
5  2019   Jan  Customer 192  Product D          21       63.660743   
6  2019   Jan  Customer 314  Product E          89       32.666768   
7  2019   Jan   Customer 59  Product B          92       97.506774   
8  2019   Jan  Customer 271  Product C          55       98.406934   
9  2019   Jan  Customer 131  Product E          51       69.629216   

       Revenue    Customer_Name  
0  2085.646368     Joshua Moore  
1   416.257188    Joseph Wilson  
2  6546.505039    Hannah Harris  
3   227.900200      Emily Davis  
4   755.166465     Rachel Brown  
5  1336.875612   Matthew Harris  
6  2907.342380  Joseph Robinson  
7  8970.623235   Chris Martinez  
8  5412.381378   David Williams  
9  3551.090024    Matthew White

***Muestra los últimos diez registros***

In [ ]:
df.tail(10)

Year Month      Customer    Product  Units_Sold  Price_per_Unit  \
4970  2023   Dec  Customer 496  Product B          98       28.855667   
4971  2023   Dec  Customer 485  Product E          52       60.084110   
4972  2023   Dec  Customer 234  Product B          67       93.580639   
4973  2023   Dec  Customer 245  Product B          32       96.596745   
4974  2023   Dec    Customer 9  Product B          57       56.436275   
4975  2023   Dec  Customer 110  Product E           8       54.706995   
4976  2023   Dec  Customer 374  Product B          82       89.649809   
4977  2023   Dec  Customer 237  Product E          10       45.605952   
4978  2023   Dec   Customer 44  Product C           8       92.443888   
4979  2023   Dec  Customer 126  Product C          64       99.183177   

          Revenue     Customer_Name  
4970  2827.855390  Michael Martinez  
4971  3124.373742    Joshua Johnson  
4972  6269.902839     Joseph Martin  
4973  3091.095834      Laura Martin  
4974  3216.867691    Chris Robinson  
4975   437.655961  Michael Williams  
4976  7351.284306        Jane Jones  
4977   456.059518     Megan Johnson  
4978   739.551104  Jessica Williams  
4979  6347.723327   Michael Johnson

***Conocer las dimensiones exactas (filas y columnas)***

In [ ]:
df.shape

(4980, 8)

***Revisar los nombres de las columnas para verificar su formato.***

In [ ]:
df.columns

Index(['Year', 'Month', 'Customer', 'Product', 'Units_Sold', 'Price_per_Unit',
       'Revenue', 'Customer_Name'],
      dtype='object')

# *Exploración y Diagnóstico*

***Diagnosticar los tipos de datos de cada columna y detectar valores nulos.***

In [ ]:
df.info() # en .info() validamos la completitud de los datos, donde nos muestra que no hay ninguna columna con datos nulos.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4980 entries, 0 to 4979
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Year            4980 non-null   int64  
 1   Month           4980 non-null   object 
 2   Customer        4980 non-null   object 
 3   Product         4980 non-null   object 
 4   Units_Sold      4980 non-null   int64  
 5   Price_per_Unit  4980 non-null   float64
 6   Revenue         4980 non-null   float64
 7   Customer_Name   4980 non-null   object 
dtypes: float64(2), int64(2), object(4)
memory usage: 311.4+ KB


In [ ]:
df.dtypes # encontramos los tipos de datos 2 enteros, 2 flotantes, 4 string

Year                int64
Month              object
Customer           object
Product            object
Units_Sold          int64
Price_per_Unit    float64
Revenue           float64
Customer_Name      object
dtype: object

***Obtener un resumen estadístico de las columnas numéricas para detectar anomalías***

In [ ]:
df.describe()

Year   Units_Sold  Price_per_Unit      Revenue
count  4980.000000  4980.000000     4980.000000  4980.000000
mean   2021.000000    49.490161       52.222311  2587.524606
std       1.414356    28.583224       27.494928  2184.779178
min    2019.000000     1.000000        5.005019    10.076287
25%    2020.000000    24.000000       28.795445   799.331333
50%    2021.000000    50.000000       52.269208  1969.560283
75%    2022.000000    74.000000       75.887524  3889.162621
max    2023.000000    99.000000       99.972226  9839.602715

***Obtener un resumen de las columnas categóricas (de texto).***

In [ ]:
df.describe(include='object')

Month      Customer    Product     Customer_Name
count   4980          4980       4980              4980
unique    12           500          5               400
top      Jan  Customer 275  Product B  Jessica Williams
freq     415            20       1062                25

***Para contar el número de valores faltantes (NaN - Not a Number) en cada columna de un DataFrame***

In [ ]:
df.isna().sum()

Year              0
Month             0
Customer          0
Product           0
Units_Sold        0
Price_per_Unit    0
Revenue           0
Customer_Name     0
dtype: int64

In [ ]:
print(df.isnull().sum())

Year              0
Month             0
Customer          0
Product           0
Units_Sold        0
Price_per_Unit    0
Revenue           0
Customer_Name     0
dtype: int64


# *Selección Inicial*

***Acceder y aislar columnas específicas por su nombre.***

In [ ]:
df['Customer_Name'] # Listar información de una columna en especifíco

0           Joshua Moore
1          Joseph Wilson
2          Hannah Harris
3            Emily Davis
4           Rachel Brown
              ...       
4975    Michael Williams
4976          Jane Jones
4977       Megan Johnson
4978    Jessica Williams
4979     Michael Johnson
Name: Customer_Name, Length: 4980, dtype: object

In [ ]:
df[['Customer_Name','Revenue']] # Mostrar registros de dos columnas

Customer_Name      Revenue
0         Joshua Moore  2085.646368
1        Joseph Wilson   416.257188
2        Hannah Harris  6546.505039
3          Emily Davis   227.900200
4         Rachel Brown   755.166465
...                ...          ...
4975  Michael Williams   437.655961
4976        Jane Jones  7351.284306
4977     Megan Johnson   456.059518
4978  Jessica Williams   739.551104
4979   Michael Johnson  6347.723327

[4980 rows x 2 columns]

In [ ]:
df[['Year','Month','Product','Revenue']] # Seleccionando varias columnas

Year Month    Product      Revenue
0     2019   Jan  Product D  2085.646368
1     2019   Jan  Product E   416.257188
2     2019   Jan  Product C  6546.505039
3     2019   Jan  Product C   227.900200
4     2019   Jan  Product D   755.166465
...    ...   ...        ...          ...
4975  2023   Dec  Product E   437.655961
4976  2023   Dec  Product B  7351.284306
4977  2023   Dec  Product E   456.059518
4978  2023   Dec  Product C   739.551104
4979  2023   Dec  Product C  6347.723327

[4980 rows x 4 columns]

***Seleccionar datos por sus etiquetas de fila y columna (nombres)***

In [ ]:
df.loc[0] # Información del registro 0

Year                      2019
Month                      Jan
Customer          Customer 103
Product              Product D
Units_Sold                  93
Price_per_Unit       22.426305
Revenue            2085.646368
Customer_Name     Joshua Moore
Name: 0, dtype: object

In [ ]:
df.loc[156] # Información del registro 156

Year                      2019
Month                      Feb
Customer          Customer 369
Product              Product A
Units_Sold                  56
Price_per_Unit       55.195162
Revenue            3090.929045
Customer_Name       John Brown
Name: 156, dtype: object

In [ ]:
df.loc[4500] # Información del registro 4500

Year                        2023
Month                        Jul
Customer            Customer 106
Product                Product A
Units_Sold                    65
Price_per_Unit         44.583658
Revenue              2897.937764
Customer_Name     Michael Martin
Name: 4500, dtype: object

In [ ]:
revenue_year = df.loc[
    (df['Year'] == 2023) & (df['Product'] == 'Product A'), # Condición para las filas
    ['Month', 'Product', 'Units_Sold', 'Revenue']       # Lista de columnas a mostrar
]

print(revenue_year.head(10))

     Month    Product  Units_Sold      Revenue
3985   Jan  Product A          81  2640.886222
3997   Jan  Product A           8   508.108325
3998   Jan  Product A          48  1887.904709
4000   Jan  Product A          63  5912.591923
4003   Jan  Product A          61  5102.879413
4009   Jan  Product A          97  4416.800731
4016   Jan  Product A          21   695.147659
4026   Jan  Product A          62  6078.786826
4032   Jan  Product A          52  2088.086698
4041   Jan  Product A          53  4050.307686


In [ ]:
product_month = df.loc[
    (df['Month'] == 'Feb'), # Condición para las filas
    ['Product', 'Price_per_Unit', 'Units_Sold', 'Revenue']       # Lista de columnas a mostrar
]

print(product_month.head(10))

      Product  Price_per_Unit  Units_Sold      Revenue
83  Product A       21.101811          67  1413.821333
84  Product B       85.200248          95  8094.023595
85  Product B       68.758634          32  2200.276291
86  Product B       33.052606          27   892.420359
87  Product A       17.979026          33   593.307874
88  Product E       81.889310          42  3439.351010
89  Product B       63.093803          25  1577.345063
90  Product B       52.644048           7   368.508336
91  Product D       11.739387          65   763.060162
92  Product D        7.623593          87   663.252620


In [ ]:
revenue_year_month = df.loc[
    (df['Year'] == 2021) & (df['Month'] == 'Jul') & (df['Product'] == 'Product C'), # Condición para las filas
    ['Year', 'Month', 'Product', 'Price_per_Unit', 'Units_Sold', 'Revenue']       # Lista de columnas a mostrar
]

print(revenue_year_month.head(10))

      Year Month    Product  Price_per_Unit  Units_Sold      Revenue
2497  2021   Jul  Product C       43.264626          30  1297.938772
2501  2021   Jul  Product C       38.117235          33  1257.868742
2510  2021   Jul  Product C       16.037155          44   705.634800
2515  2021   Jul  Product C       90.190922           1    90.190922
2516  2021   Jul  Product C        5.517152          29   159.997419
2519  2021   Jul  Product C       57.955351          87  5042.115515
2522  2021   Jul  Product C       99.895946          29  2896.982448
2523  2021   Jul  Product C       96.267901          65  6257.413592
2525  2021   Jul  Product C       90.637449          14  1268.924283
2550  2021   Jul  Product C       84.842061          33  2799.788023


***Seleccionar datos por su posición numérica (índice).***

In [ ]:
df.iloc[0, 2:7]

Customer          Customer 103
Product              Product D
Units_Sold                  93
Price_per_Unit       22.426305
Revenue            2085.646368
Name: 0, dtype: object

In [ ]:
df.iloc[1500, 0:8]

Year                      2020
Month                      Jul
Customer           Customer 35
Product              Product E
Units_Sold                  26
Price_per_Unit       88.059197
Revenue            2289.539115
Customer_Name     Sophia Davis
Name: 1500, dtype: object

In [ ]:
df.iloc[3, 0:-1]

Year                      2019
Month                      Jan
Customer          Customer 100
Product              Product C
Units_Sold                  22
Price_per_Unit         10.3591
Revenue               227.9002
Name: 3, dtype: object

# *Cuestionario integrado (responder al final del código)*

1. Dimensiones totales: ¿Cuántas filas y columnas contiene df?

R/= Contiene 4980 Filas y 8 columnas.

2. Nulos globales: ¿Cuántas celdas nulas hay en todo el DataFrame y qué porcentaje representan sobre el total de celdas?

R/= No hay valores nulos 0 en cantidad y 0%.

3. Columna más problemática: Indique la columna con mayor número absoluto de nulos y el porcentaje de nulos que presenta esa columna respecto a su propio total de filas.

R/= Ninguna columna cuenta con valores nulos por lo que no hay un a columna problemática.

4. Variabilidad numérica: Entre las columnas numéricas, identifique la que posee la mayor desviación estándar. Anote su valor mínimo, máximo y desviación estándar.

R/= La Columna Revenue cuenta con mayor valor de desviación estandard.
*   Mínimo: 10.076287
*   Máximo: 9839.602715
*   Desviación estandard: 2184.779178

5. Variedad categórica: Entre las columnas de tipo object, ¿cuál contiene el mayor número de categorías únicas? ¿Cuántas categorías son?

R/= La columna que contiene el mayor número de categorías únicas es la columna Customer con un valor de 500 valores únicos.